### PR TITLE
Adding support for .env files for cppvsdbg

### PR DESCRIPTION
**Pull request by WardenGnaw**
This PR allows users to have a .env file that listed as
KEY=VALUE or KEY="VALUE WITH SPACE" to be used as environment variables
when launching a process. Anything that begins with '#' is treated as a
comment. Blank lines are ignored.

This .env file will be used if envFile passed in with the launch.json.
It will also work with the existing environment variable, however
anything listed in the .env file will overwrite anything from the
environment field in the launch.json.

@gregg-miskelly

### CODE REVIEW

**Comment by WardenGnaw** [comment]
Addresses https://github.com/Microsoft/vscode-cpptools/issues/3490

**Comment by michelleangela** [comment]
Is this targeting 0.23.0-insiders?

File: Extension/package-lock.json [hunk 1]
```diff
@@ -1569,7 +1569,8 @@
         "ansi-regex": {
           "version": "2.1.1",
           "bundled": true,
-          "dev": true
+          "dev": true,
+          "optional": true
```
Comment (sean-mcmanus) [inline Extension/package-lock.json]: Why did this change?

Comment (WardenGnaw) [reply Extension/package-lock.json]: No idea.

**Comment by WardenGnaw** [comment]
@michelleangela there’s no big rush for this PR

**Review by chuckries — APPROVED** [review]

File: Extension/package.json [hunk 2]
```diff
@@ -1201,6 +1201,11 @@
                 },
                 "default": []
               },
+              "envFile": {
+                "type": "string",
+                "description": "Environment variables passed to the program by a file.",
```
Comment (bobbrow) [inline Extension/package.json]: Should there be more description about the format of the file here, or are users expected to know what this file is?

Comment (WardenGnaw) [reply Extension/package.json]: We will need to update the vscode docs or add to launch.md with a similar description in https://code.visualstudio.com/docs/python/environments#_environment-variable-definitions-file.

Any preference?

Comment (bobbrow) [reply Extension/package.json]: No preference. If you move it to the vscode docs, you may want to mention that in the description here.

Comment (WardenGnaw) [reply Extension/package.json]: Added to launch.md

**Comment by sean-mcmanus** [comment]
@WardenGnaw Can you check this in by our 0.23.0-insiders2 on Friday? We want the next Insiders to be as close to the final release as possible.

**Pull request merged**